src/contracts/pods/EigenPod.sol

```diff
@@ -488,274 +451,182 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
-         * (WITHDRAWN is allowed because technically you can deposit to a validator even after it exits)
-         */
+    /**
+     * @dev Initiate a checkpoint proof by snapshotting both the pod's ETH balance and the
+     * current block's parent block root. After providing a checkpoint proof for each of the
+     * pod's ACTIVE validators, the pod's ETH balance is awarded shares and can be withdrawn.
+     * @dev ACTIVE validators are validators with verified withdrawal credentials (See
+     * `verifyWithdrawalCredentials` for details)
+     * @dev If the pod does not have any ACTIVE validators, the checkpoint is automatically
+     * finalized.
+     * @dev Once started, a checkpoint MUST be completed! It is not possible to start a
+     * checkpoint if the existing one is incomplete.
+     * @param revertIfNoBalance If the available ETH balance for checkpointing is 0 and this is
+     * true, this method will revert
+     */
+    function _startCheckpoint(bool revertIfNoBalance) internal {
         require(
-            _validatorPubkeyHashToInfo[validatorPubkeyHash].status != VALIDATOR_STATUS.INACTIVE,
-            "EigenPod._verifyAndProcessWithdrawal: Validator never proven to have withdrawal credentials pointed to this contract"
+            currentCheckpointTimestamp == 0, 
+            "EigenPod._startCheckpoint: must finish previous checkpoint before starting another"
         );
 
-        // Ensure we don't process the same withdrawal twice
         require(
-            !provenWithdrawal[validatorPubkeyHash][withdrawalTimestamp],
-            "EigenPod._verifyAndProcessWithdrawal: withdrawal has already been proven for this timestamp"
+            lastCheckpointTimestamp != uint64(block.timestamp),
+            "EigenPod._startCheckpoint: cannot checkpoint twice in one block"
         );
 
-        provenWithdrawal[validatorPubkeyHash][withdrawalTimestamp] = true;
+        // Snapshot pod balance at the start of the checkpoint, subtracting pod balance that has
+        // previously been credited with shares. Once the checkpoint is finalized, `podBalanceGwei` 
+        // will be added to the total validator balance delta and credited as shares.
+        // 
+        // Note: On finalization, `podBalanceGwei` is added to `withdrawableRestakedExecutionLayerGwei`
+        // to denote that it has been credited with shares. Because this value is denominated in gwei, 
+        // `podBalanceGwei` is also converted to a gwei amount here. This means that any sub-gwei amounts 
+        // sent to the pod are not credited with shares and are therefore not withdrawable. 
+        // This can be addressed by topping up a pod's balance to a value divisible by 1 gwei.
+        uint64 podBalanceGwei = 
+            uint64(address(this).balance / GWEI_TO_WEI) - withdrawableRestakedExecutionLayerGwei;
+
+        // If the caller doesn't want a "0 balance" checkpoint, revert
+        if (revertIfNoBalance && podBalanceGwei == 0) {
```

Comment:
require

Reply:
`require(!revertIfNoBalance || podBalanceGwei != 0)`? or something?

idk i think the if statement is wayyy easier to understand